pkg/sensors/base/base.go

```diff
@@ -77,6 +84,43 @@ var (
 	ErrMetricsMap = program.MapBuilder(errmetrics.MapName, Execve)
 )
 
+func readThreadsMax(path string) (int64, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return 0, err
+	}
+	str := strings.TrimRight(string(data), "\n")
+	return strconv.ParseInt(str, 10, 32)
+}
+
+func GetExecveMapEntries(str string) int {
+	entry := int(unsafe.Sizeof(execvemap.ExecveValue{}))
+
+	if len(str) == 0 {
+		return ExecveMapMaxEntries
+	}
+	// pure number of entries
+	if val, err := strconv.ParseUint(str, 10, 32); err == nil {
+		return int(val)
+	}
+	// follow threads-max entries
+	if str == "max" {
+		if val, err := readThreadsMax("/proc/sys/kernel/threads-max"); err == nil {
+			return int(val)
+		}
+		logger.GetLogger().Warn("Failed to read /proc/sys/kernel/threads-max file, falling back to default")
+		return ExecveMapMaxEntries
+	}
+	// set entries based on size
+	size, err := strutils.ParseSize(str)
```

Comment:
Do I understand correctly that we set entries based on size if there is a suffix? For example, `1024` is interpreted as the number of entries, but `1K` would be interpreted as the total size of the map?

It seems that this could cause confusion. An alternative would be to use two options, such as `--execve-map-max-entries` and `--execve-map-max-size`, and just keep one if both are set (maybe with a warning).